lib/pbench/test/unit/agent/task/test_copy_result_tb.py

```diff
@@ -151,20 +155,22 @@ def test_connection_error(self, monkeypatch, agent_logger):
             responses.PUT, upload_url, body=requests.exceptions.ConnectionError("uh-oh")
         )
 
-        monkeypatch.setattr(Path, "exists", self.get_path_exists_mock(tb_name, True))
-        monkeypatch.setattr(
-            Path, "open", self.get_path_open_mock(tb_name, io.StringIO(tb_contents))
-        )
-
-        with pytest.raises(RuntimeError) as excinfo:
-            crt = CopyResultTb(
-                tb_name,
-                len(tb_contents),
-                "someMD5",
-                self.config,
-                agent_logger,
+        with monkeypatch.context() as m:
+            m.setattr(Path, "exists", self.get_path_exists_mock(tb_name, True))
+            m.setattr(
+                Path, "open", self.get_path_open_mock(tb_name, io.StringIO(tb_contents))
             )
-            crt.copy_result_tb("token")
+
+            with pytest.raises(RuntimeError) as excinfo:
+                crt = CopyResultTb(
+                    tb_name,
+                    len(tb_contents),
+                    "someMD5",
+                    self.config,
+                    agent_logger,
+                )
+                res = crt.copy_result_tb("token")
+                assert not res.ok
```

Comment:
The `assert` at line 173 is never executed, right?...because either the call to `CopyResultTb()` or the call to `copy_result_tb()` is going to raise `RuntimeError`...and, if it doesn't, then per line 164, PyTest will report a test failure.  So, I think you should omit line 173.

Ditto line 202.

Reply:
`flake8` complained about the unused variable when I took them out, so I just left them in and added comments.

Reply:
I can't remember where the `RuntimeError` is raised.  I assume that it comes from the call to `copy_result_tb()`, otherwise, you can/should remove that call.  Given that it comes from there, there is no purpose for the test to capture its return value in `res` (because execution will abort before a value is returned) and so you should keep the call but omit the assignment (which will pacify Flake8), and then you should remove the assertion (which will pacify me 😉).